lib/helpers/DocumentTypeApiHelper.ts

```diff
@@ -597,4 +597,34 @@ export class DocumentTypeApiHelper {
       .build();
     return await this.create(documentType);
   }
+
+  async createDocumentTypeWithPropertyEditorAndAllowedTemplate(documentTypeName: string, dataTypeId: string, propertyName: string, templateId: string) {
+    const crypto = require('crypto');
+    const containerId = crypto.randomUUID();
+    await this.ensureNameNotExists(documentTypeName);
+
+    const documentType = new DocumentTypeBuilder()
+      .withName(documentTypeName)
+      .withAlias(AliasHelper.toAlias(documentTypeName))
+      .withAllowedAsRoot(true)
+      .addContainer()
+        .withName('TestGroup')
+        .withId(containerId)
+        .withType('Group')
+        .done()
+      .addProperty()
+        .withContainerId(containerId)
+        .withAlias(AliasHelper.toAlias(propertyName))
+        .withName(propertyName)
+        .withDataTypeId(dataTypeId)
+        .done()
+      .addAllowedTemplateId()
+        .withId(templateId)
+        .done()
+      .withDefaultTemplateId(templateId)
+      .build();
+    
+    const documentTypeId =  await this.create(documentType);
+    return documentTypeId === undefined ? '' : documentTypeId;
```

Comment:
Is this necessary?

Reply:
Yes, the output of this function is the input for the createPublishedDocumentWithValue function, and in that function, documentTypeId should not be undefined. To ensure consistency in the code, I will move this logic from the current function to the createPublishedDocumentWithValue function.